pynamodb/models.py

```diff
@@ -542,16 +542,24 @@ def count(cls,
             non_key_attribute_classes=non_key_attribute_classes,
             filters=filters)
 
-        data = cls._get_connection().query(
-            hash_key,
-            index_name=index_name,
-            consistent_read=consistent_read,
-            key_conditions=key_conditions,
-            query_filters=query_filters,
-            limit=limit,
-            select=COUNT
-        )
-        return data.get(CAMEL_COUNT)
+        count_buffer = 0
+        last_evaluated_key = None
+        data = None
+        while not data or last_evaluated_key:
```

Comment:
probably a stupid question but because I'm not aware of the output of query I'll ask it anyway - will query always return an object?  I'm wondering about the case where there are no items and I want to make sure this won't loop forever because of that "not data".